### ISSUE TITLE
Mission generation failure "could not find a carrier in the mission matching..."

### ISSUE THREAD

**Issue by kahssymhir** [dcs-liberation/dcs_liberation#3162]
### Affected versions

8.1.0

### Description

I can no longer generate a mission with the "take-off" button on Turn 3 and 4 of a Persian gulf campaign due to the following error. It seems the position of the carrier is problematic, I am not sure if it is because of the proximity to the dead enemy boats or proximity to the shore, the latter never having caused such an issue. Passing the turn without changing the position of the carrier does not fix the issue. The issue disappeared once I order the boat to sail away from the dead enemy boats at the next turn. 

![image](https://github.com/dcs-liberation/dcs_liberation/assets/143124440/0eda7241-1442-474b-87e6-6c2b697a00a4)

Steps to reproduce (as far as I know)

1. Bring your carrier in the same location as on the picture.

![image](https://github.com/dcs-liberation/dcs_liberation/assets/143124440/064c835b-097d-4ff3-be82-61cc810d745d)

3. Generate flight plans
4. Press take-off

Thanks, I hope it helps. 

### Save game and other files (save game required, bugs without saves will be closed)

[Iran Invasion.liberation.zip](https://github.com/dcs-liberation/dcs_liberation/files/12432472/Iran.Invasion.liberation.zip)

**Comment by kahssymhir** [comment]
It's not just about the ship position. After I confirmed that the new ship position would work, I redid all my packages and once all done the error came back again. I tried with a few packages - no issue.  but after redoing all my packages the error came back. See attached.  
[Iran Invasion 2.liberation.liberation.zip](https://github.com/dcs-liberation/dcs_liberation/files/12433238/Iran.Invasion.2.liberation.liberation.zip)
